.github/workflows/publish.yml

```diff
@@ -0,0 +1,116 @@
+        required: true
+        type: string
+      no-intl:
+        description: 'Build with --with-intl=none'
+        required: false
+        type: boolean
+        default: false
+
+env:
+  DISTRO: ${{ inputs.distro || 'debian' }}
+  ARCH: ${{ inputs.arch || 'aarch64' }}
+  NODE_VERSION: ${{ inputs.node-version || 'v19.6.1' }}
+  NONE_INTL: ${{ inputs.no-intl || '' }}
+
+permissions:
+  actions: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: write # AWS GitHub OIDC required: write
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  pull-requests: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  publish:
+    name: Publish binaries
+    runs-on: ubuntu-latest
+
+    steps:
+      # https://github.com/actions/checkout
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+
+      # https://github.com/aws-actions/configure-aws-credentials
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+        id: aws-credentials
+        with:
+          role-to-assume: ${{ vars.AWS_IAM_ROLE || 'arn:aws:iam:491725000532:user/balena-packages-user' }}
+          role-session-name: github-${{ github.job }}-${{ github.run_id }}-${{ github.run_attempt }}
+          aws-region: ${{ vars.AWS_REGION || 'us-east-1' }}
+          # https://github.com/orgs/community/discussions/26636#discussioncomment-3252664
+          mask-aws-account-id: false
+
+      # TODO: use self-hosted amd64/arm64 runners instead of QEMU
+      # https://github.com/docker/setup-qemu-action
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3.0.0
+        env:
+          LOG_LEVEL: debug
+          # renovate: datasource=docker depName=binfmt packageName=tonistiigi/binfmt
+          BINFMT_VERSION: qemu-v8.0.4-33
+        with:
+          # https://hub.docker.com/r/tonistiigi/binfmt
+          image: tonistiigi/binfmt:${{ env.BINFMT_VERSION }}
+
+      # FIXME: this is insecure as the helper image is built from untrusted user code
+      - name: Build helper image
+        working-directory: ${{ env.DISTRO }}/${{ env.ARCH }}
+        run: |
+          cp -f ../../commit-table .
+          sed -e s~#{SUITE}~buster~g Dockerfile.tpl > Dockerfile
+          docker build --no-cache=true -t "node-$DISTRO-$ARCH-builder" .
+
+      # TODO: are AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY set by the above action?
+      - name: Build and publish binaries
+        # # do not deploy on pull requests
+        # if: github.event_name != 'pull_request'
+        working-directory: ${{ env.DISTRO }}/${{ env.ARCH }}
+        env:
+          BUCKET_NAME: ${{ vars.BUCKET_NAME || 'balena-packages' }}
+          ACCESS_KEY: ${{ steps.aws-credentials.outputs.aws-access-key-id }}
+          SECRET_KEY: ${{ steps.aws-credentials.outputs.aws-secret-access-key }}
```

Comment:
We need to set `AWS_SESSION_TOKEN` to `${{ steps.creds.outputs.aws-session-token }}` since we are using temporary credentials from an IAM role.  We may also want to align the environment variable names with those [used by AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html).